src/ralph/backends/data/ldp.py

```diff
@@ -214,24 +214,10 @@ def read(
             }
         )
 
-    def write(  # pylint: disable=too-many-arguments
-        self,
-        data: Iterable[Union[bytes, dict]],
-        target: Optional[str] = None,
-        chunk_size: Optional[int] = None,
-        ignore_errors: bool = False,
-        operation_type: Optional[BaseOperationType] = None,
-    ) -> int:
-        """LDP data backend is read-only, calling this method will raise an error."""
-        msg = "LDP data backend is read-only, cannot write to %s"
-        logger.error(msg, target)
-        raise NotImplementedError(msg % target)
-
     def close(self) -> None:
-        """LDP client does not support close, this method is not implemented."""
-        msg = "LDP data backend does not support `close` method"
-        logger.error(msg)
-        raise NotImplementedError(msg)
+        """LDP data backend has no open connections to close. No action."""
+        self._client = None
+        logger.info("No open connections to close; skipping")
```

Comment:
Maybe I missed something, but it seems to me like the initial version is clearer (including raising `NotImplementedError`)

Reply:
Good point) In this case, when the backend inherits an abstract ABC interface, it's like promising that all abstract methods are implemented.
When we implement an abstract method and raise a `NotImplementedError`, we somehow break this promise.

The goal here is to make things smoother for end-users. By simply knowing the abstract interface, they should be able to use any backend without worrying about which methods are implemented :

```python
backend = some_backend() # might raise NotImplementedError
backend.read()  #  might raise BackendException or BackendParameterException
backend.close()  #  might raise BackendException
```